### PR TITLE
Navigation Component: Update docs

### PR DESCRIPTION
**Pull request by psealock**
Fixes https://github.com/WordPress/gutenberg/issues/24700

## Description

Update the navigation component's README and other docs.

### CODE REVIEW

File: packages/components/src/navigation/README.md [hunk 1]
```diff
@@ -1,26 +1,74 @@
...
 
-Navigation is a component which renders a component which renders a heirarchy of menu items.
+Render a heirarchy of menu items into a waterfall style navigation.
 
 ## Usage
 
 ```jsx
-import { MenuItem } from '@wordpress/components';
-import { withState } from '@wordpress/compose';
+import Navigation from '@wordpress/components';
```
Comment (joshuatf) [inline packages/components/src/navigation/README.md:8]: I don't think we can import from the default and the names may include "experimental" for a while; not sure if worth noting here:

```suggestion
import {
	__experimentalNavigation as Navigation,
	__experimentalNavigationMenu as NavigationMenu,
	__experimentalNavigationMenuItem as NavigationMenuItem,
} from '@wordpress/components';
```

File: packages/components/src/navigation/README.md [hunk 2]
```diff
@@ -1,26 +1,74 @@
 # Navigation
 
-Navigation is a component which renders a component which renders a heirarchy of menu items.
+Render a heirarchy of menu items into a waterfall style navigation.
```
Comment (joshuatf) [inline packages/components/src/navigation/README.md:3]: Great description.  One small typo ("heirarchy") and optional, but should we include "flat" to help signal the data structure used?

```suggestion
Render a flat array of menu items into a waterfall style hierarchy navigation.
```

I'm not sure if that makes things more clear or not; will leave that up to your discretion.

File: packages/components/src/navigation/README.md [hunk 3]
```diff
@@ -30,14 +78,65 @@ Navigation supports the following props.
...
+-   Required: Yes
+
+A menu item's id.
+
+#### `config.parent`
+
+-   Type: `string|Number`
+-   Required: No
+
+Specify a menu item's parent id.
```
Comment (joshuatf) [inline packages/components/src/navigation/README.md:107]: Maybe note that `root` will get added as the menu item's parent if none is provided.

File: packages/components/src/navigation/README.md [hunk 4]
```diff
@@ -30,14 +78,65 @@ Navigation supports the following props.
...
 
-- Type: `array`
-- Required: Yes
+-   Type: `array`
+-   Required: Yes
 
 An array of config objects for each menu item.
 
-### `initial`
+Config objects can be represented
```
Comment (joshuatf) [inline packages/components/src/navigation/README.md:86]: Missing `onClick` param option.

File: packages/components/src/navigation/README.md [hunk 5]
```diff
@@ -30,14 +78,65 @@ Navigation supports the following props.
...
+-   Required: No
+
+Turn a menu item into a link by supplying a url.
+
+#### `config.linkProps`
+
+-   Type: `object`
+-   Required: No
+
+Supply an properties passed to the menu-item.
```
Comment (joshuatf) [inline packages/components/src/navigation/README.md:121]:
```suggestion
Supply properties passed to the menu-item.
```

**Review by joshuatf — APPROVED** [review]
Thanks for the docs, Paul!  Left a couple minor and optional changes, but pre-approving. 🎉

File: packages/components/src/navigation/README.md (hunk 1 above)
Comment (psealock) [reply packages/components/src/navigation/README.md:8]: Probably a good idea. There are instances of each present in the codebase.

https://github.com/WordPress/gutenberg/blob/ebd5d9605f0aa57088f74d0ecaccf93017e36d5e/packages/components/src/number-control/README.md#L8

https://github.com/WordPress/gutenberg/blob/ebd5d9605f0aa57088f74d0ecaccf93017e36d5e/packages/components/src/text/README.md#L8

**Pull request merged**